### PR TITLE
Fixed issue #17883: QueXML formatting cannot be changed (master)

### PR DESCRIPTION
**Pull request by adamzammit**
Fixed issue #17883: : QueXML formatting cannot be changed (master)
Dev: This is a similar fix for master that was merged in to 3.x-LTS (see #2561 )

### CODE REVIEW

**Review by gabrieljenik — APPROVED** [review]
Code looks OK. Haven't tested it.

**Comment by adamzammit** [comment]
I've just rebased to 6.0.0 and needed to make some changes. I have tested this now on 6.0.0

File: application/controllers/PrintanswersController.php [hunk 1]
```diff
@@ -185,6 +185,15 @@ public function actionView($surveyid, $printableexport = false)
 
             $quexmlpdf = new quexmlpdf();
 
+            //apply settings stored at last output
```
Comment (olleharstedt) [inline application/controllers/PrintanswersController.php:188]: Code duplication? Looks like the same snippet as below?

**Review by olleharstedt — CHANGES_REQUESTED** [review]

File: application/controllers/PrintanswersController.php (hunk 1 above)
Comment (gabrieljenik) [reply application/controllers/PrintanswersController.php:188]: Yes, could be a routine of quexml?
`applyGlobalSettings()` ?

Comment (adamzammit) [reply application/controllers/PrintanswersController.php]: thanks for the suggestion I've done that

File: application/libraries/admin/quexmlpdf.php [hunk 2]
```diff
@@ -2260,6 +2260,22 @@ public function create($questionnaire)
         $this->fillLastPageBackground();
     }
 
+    /**
+     * Apply global settings from LimeSurvey application
+     *
+     * @access public
```
Comment (olleharstedt) [inline application/libraries/admin/quexmlpdf.php:2266]: This annotation is not needed, it's already set to `public` with the keyword. :)

File: application/libraries/admin/quexmlpdf.php [hunk 3]
```diff
@@ -2260,6 +2260,22 @@ public function create($questionnaire)
...
+     */
+    public function applyGlobalSettings()
```
Comment (olleharstedt) [inline application/libraries/admin/quexmlpdf.php:2268]: Nice!

File: application/libraries/admin/quexmlpdf.php [hunk 4]
```diff
@@ -2260,6 +2260,22 @@ public function create($questionnaire)
...
+    {
+        foreach ($this->_quexmlsettings() as $s) {
+            $setting = App()->getConfig($s);
+                if ($setting !== null && trim($setting) !== '') {
```
[the remark this answers is not in the document]
Comment (adamzammit) [reply application/libraries/admin/quexmlpdf.php]: thanks I've fixed this

File: application/libraries/admin/quexmlpdf.php (hunk 2 above)
Comment (adamzammit) [reply application/libraries/admin/quexmlpdf.php]: removed :)

**Pull request merged**